### PR TITLE
Python 3 + Mixbox Support

### PR DESCRIPTION
**Pull request by gtback**
This obsoletes #228 and incorporates mixbox, with the common "binding utils" delegated to that package. Mixbox natively supports Python 3, and this branch has also been updated to support Python 3.2, 3.3, and 3.4.

This requires mixbox 0.0.2. I didn't explicitly specify a version, since I don't want to remember to bump it each time in several places. If you have a different version of mixbox cached locally with pip, you can just "pip install mixbox==0.0.2" from any virtualenv to download and cache the newest version, which will get used to build new venvs.

### CODE REVIEW

**Comment by gtback** [comment]
@bworrell, I know this is a huge change, but do you have time to look at it and provide any feedback?

File: cybox/bindings/win_task_object.py [hunk 1]
```diff
@@ -1523,7 +1522,7 @@ def buildChildren(self, child_, node, nodeName_, fromsubclass_=False):
     'Fuzzy_Hash_Value': cybox_common.FuzzyHashValueType,
     'Show_Message_Title': cybox_common.StringObjectPropertyType,
     'Data_Size': cybox_common.DataSizeType,
-    'Email_Message': email_message_object.EmailMessageObjectType,
+    'email_message': email_message_object.EmailMessageObjectType,
```
Comment (bworrell) [inline cybox/bindings/win_task_object.py]: What's this for?

File: cybox/common/structured_text.py [hunk 2]
```diff
@@ -76,7 +79,4 @@ def from_dict(cls, text_dict, text=None):
         return text
 
     def __str__(self):
-        return self.__unicode__().encode("utf-8")
-    
-    def __unicode__(self):
-        return unicode(self.value)
+        return self.value
```
Comment (bworrell) [inline cybox/common/structured_text.py]: Does this need to be wrapped in a `six.text_type(...)` call?

File: cybox/objects/artifact_object.py [hunk 3]
```diff
@@ -53,17 +75,22 @@ def data(self):
...
             raise ValueError("packed_data already set, can't set data")
+        if value is not None and not isinstance(value, six.binary_type):
+            msg = ("Artifact data must be either None or byte data, not a "
+                   "Unicode string.")
+            raise ValueError(msg)
         self._data = value
 
     @property
     def packed_data(self):
+        "Should return a Unicode string"
```
Comment (bworrell) [inline cybox/objects/artifact_object.py]: This should be triple-quoted.

File: cybox/test/__init__.py [hunk 4]
```diff
@@ -52,8 +54,8 @@ def round_trip(o, output=False, list_=False):
 
     klass = o.__class__
     if output:
-        print "Class: ", klass
-        print "-" * 40
+        print("Class: ", klass)
```
Comment (bworrell) [inline cybox/test/__init__.py]: Not needed for this PR, but I'm wondering if we should switch to using `logging` calls over `print` statements.

File: cybox/test/encoding_test.py [hunk 5]
```diff
@@ -53,85 +56,6 @@ def test_whois(self):
         o.dnssec = UNICODE_STR
         o2 = round_trip(o)
 
-    def test_quote_xml(self):
```
Comment (bworrell) [inline cybox/test/encoding_test.py]: I had a moment of panicked confusion before realizing that these tests were moved to `mixbox` :)

**Comment by bworrell** [comment]
@gtback, I looked through the code and posted a few comments/questions but overall I think it looks awesome! 

I ran `nosetests` against my local python-stix clone and it failed because `ExternalEncoding` no longer lives in `cybox.bindings`. If I add the following line to `__init__.py` in `cybox.bindings`, everything passes:

```python
# For python-stix/python-maec
from mixbox.binding_utils import ExternalEncoding
```

I'm not sure if we want to add the line, or just work towards getting python-stix|maec up and running with `mixbox` before pushing out a new release of python-cybox. What do you think?

File: cybox/bindings/win_task_object.py (hunk 1 above)
Comment (gtback) [inline cybox/bindings/win_task_object.py]: I think it is a find-replace gone bad. It pops up in a couple spots. I fixed it in 8327de6e1f60b32050910abcba88d24e6bd87e5f.

File: cybox/test/__init__.py (hunk 4 above)
Comment (gtback) [inline cybox/test/__init__.py]: Yeah, the "output" flag is typically off; it's only there to add during development, so you can actually see the output at each stage of the round trip. Ultimately these "testing utilities" should move to mixbox, we can fix it there. Does logging output show up in test failures?

**Comment by gtback** [comment]
I would lean towards not adding to `cybox.bindings.__init__`, since the python-stix|maec code doesn't actually do what it's supposed to after that change (it redefines `cybox.bindings.ExternalEncoding`, but that doesn't actually modify `mixbox.binding_utils.ExternalEncoding` which is what the CybOX bindings use for output). Better to just update python-stix|maec quickly. I don't see an immediate need for a python-cybox release.

File: cybox/test/__init__.py (hunk 4 above)
Comment (bworrell) [inline cybox/test/__init__.py]: Cool cool. Nose has a lot of [logging options](http://nose.readthedocs.org/en/latest/plugins/logcapture.html) and by default displays log messages on error.

**Comment by bworrell** [comment]
@gtback, this looks awesome! I'm gonna merge, though we'll need to update python-stix and python-maec before the next release. Thanks!!

**Pull request merged**